indexer/consensus_client.go

```diff
@@ -177,6 +189,19 @@ func (client *ConsensusClient) checkClient() error {
 		return fmt.Errorf("initialization of attestantio/go-eth2-client failed: %w", err)
 	}
 
+	// get peerId
+	client.peerId, err = client.rpcClient.GetNodePeerId()
+	if err != nil {
+		return fmt.Errorf("could not get node peer id: %v", err)
+	}
+
+	// get peers
+	peers, err := client.rpcClient.GetNodePeers()
+	if err != nil {
+		return fmt.Errorf("could not get peers: %v", err)
+	}
+	client.peers = peers
```

Comment:
the peer list loaded in here is not getting refreshed for ready clients as the whole `checkClient()` function is only called for offline clients to check if the client is ready to be used as data source.
once clients are ready, the execution is "locked" inside the `processClientEvents()` "process events" loop.

I think you should still keep it there to refresh the clients list for syncing clients, but also add the refresh to the process events loop.

Reply:
I've made the peer check happen during epoch changes as per recommendation to not query it too often.